src/extension.c

```diff
@@ -62,7 +64,20 @@ static enum ExtensionState extstate = EXTENSION_STATE_UNKNOWN;
  * often need it during the planning, so we cache it here. We update it when
  * the extension status is updated.
  */
-Oid ts_extension_oid = InvalidOid;
+static Oid ts_extension_oid = InvalidOid;
+
+Oid
+ts_extension_get_oid(void)
+{
+	if (OidIsValid(ts_extension_oid))
+	{
+		Assert(ts_extension_oid == get_extension_oid(EXTENSION_NAME, true));
```

Comment:
This might trigger problems in debug builds.

Reply:
Just to clarify: this might trigger complicated issues in a debug build that are non-existent in the real build and hence will cause waste of time on non-existent issues. It is probably better to not do this at all, or do it in a location that is not called from invalidation code.

Reply:
Should not be a problem since the function is not called from invalidation callback code.

Reply:
Is it possible to add a check for that?

Reply:
Do you suggest checking that this function is not called from an invalidation callback? 

That makes no sense, IMO. Why single out this function when any function that calls the relcache leads to bad behavior?

We'd need something that generally checks if a relcache function is called from an invalidation callback, and I don't see how that can be done.

Reply:
> Do you suggest checking that this function is not called from an invalidation callback?

Not suggesting anything, just wondering if there is an easy way to add a check and if it is meaningful. I guess from the rest of your reply that the answer is "no".